src/pybind/mgr/dashboard/frontend/src/app/ceph/rgw/models/rgw-multisite-zone-deletion-form/rgw-multisite-zone-deletion-form.component.ts

```diff
@@ -57,4 +66,33 @@ export class RgwMultisiteZoneDeletionFormComponent implements OnInit {
   showDangerText() {
     this.displayText = !this.displayText;
   }
+
+  updateIncludedPools(): void {
+    if (!this.zoneData$ || !this.poolList$) {
+      return;
+    }
+
+    this.zoneData$.subscribe((data: any) => {
+      this.poolList$.subscribe((poolList: any) => {
+        for (let pool of poolList) {
+          for (let zonePool of Object.values(data)) {
+            if (
+              typeof zonePool === 'string' &&
+              zonePool.includes(pool.pool_name) &&
+              !this.includedPools.includes(pool.pool_name)
+            ) {
+              this.includedPools.push(pool.pool_name);
+            } else if (
+              Array.isArray(zonePool) &&
+              zonePool[0].val.storage_classes.STANDARD.data_pool &&
+              zonePool.includes(pool.pool_name) &&
+              !this.includedPools.includes(zonePool[0].val.storage_classes.STANDARD.data_pool)
+            ) {
+              this.includedPools.push(zonePool[0].val.storage_classes.STANDARD.data_pool);
+            }
+          }
+        }
+      });
+    });
```

Comment:
```suggestion
    this.zoneData$.subscribe((data: any) => {
      this.poolList$.subscribe((poolList: any) => {
        for (let pool of poolList) {
          for (let zonePool of Object.values(data)) {
            if (
              typeof zonePool === 'string' &&
              zonePool.includes(pool.pool_name) 
            ) {
              this.includedPools.push(pool.pool_name);
            } else if (
              Array.isArray(zonePool) &&
              zonePool[0].val.storage_classes.STANDARD.data_pool &&
              zonePool.includes(pool.pool_name) 
            ) {
              this.includedPools.push(zonePool[0].val.storage_classes.STANDARD.data_pool);
            }
          }
        }
      });
    });
```
1. why is only the first value in the array checked? `zonePool[0].val.storage_classes.STANDARD.data_pool`
2. Can't we make the backend always return the same type? It looks like the string could've been converted to the array type below instead of having an extra check in the fronted.
3. If you add a set for `includedPools` you can remove those inclusion tests.

Reply:
Yep, fixed it.
Covering all values of array now, changed to a set